lib/manager/actions/deployments.js

```diff
@@ -103,6 +103,37 @@ export function deployToTarget (deployment: Deployment, target: string) {
   }
 }
 
+export function updatePelias (deployment: Deployment, resetDb?: boolean) {
+  return async function (dispatch: dispatchFn, getState: getStateFn) {
+    if (resetDb) {
+      await dispatch(updateDeployment(deployment, {peliasResetDb: true}))
+    }
+    const {id} = deployment
+    return dispatch(secureFetch(`${DEPLOYMENT_URL}/${id}/updatepelias`, 'post'))
+      .then(response => {
+        // FIXME: Use standard handleJobResponse for deployment job
+        if (response.status >= 400 || response.status === 202) {
+          // If there is an issue with the deployment, return the JSON response
+          // (handled be below alert)
+          return response.json()
```

Comment:
This might read a little better if you used async/await so the flow doesn't get interrupted.

Reply:
I agree, but this block was "adapted" from `deployToTarget`, and refactoring it led to some strange behavior. I think refactoring this is the right thing to do, but think it will require more changes elsewhere -- my guess is that the fact that the dispatch itself is being returned (which I guess means the method returns a promise?) makes using async/wait difficult.

Reply:
Not sure why `deployToTarget` originally returns a `dispatch` function. As written, `updatePelias` should be already be callable from other actions, so there should be no need to return a value.

Reply:
Have removed all return values in d54487d0. However, return is still used to jump to the second then block